external/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+# ---------------------------------------------------------------
+# SUNDIALS Copyright Start
+# Copyright (c) 2002-2024, Lawrence Livermore National Security
+# and Southern Methodist University.
+# All rights reserved.
+#
+# See the top-level LICENSE and NOTICE files for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+# SUNDIALS Copyright End
+# ---------------------------------------------------------------
+
+# Steps to add an external addon to SUNDIALS:
+#   WARNING: external addons are not maintained by the SUNDIALS team
+#   and may come with their own license with different rules.
+#
+#   1. Clone/copy the addon into <sundials root>/external/
+#      (the same directory as this file you are reading).
+#   2. Copy the sundials-addon-example block for your own external project.
+#   3. When building SUNDIALS, set the CMake option SUNDIALS_ENABLE_EXTERNAL_ADDONS to ON
+#   4. Build SUNDIALS as usual.
+
+include(FetchContent)
+
+# COPY THE BLOCK BELOW FOR YOUR OWN EXTERNAL ADDON
+FetchContent_Declare(sundials-addon-example
+  SOURCE_DIR ${PROJECT_SOURCE_DIR}/external/sundials-addon-example
+)
+FetchContent_MakeAvailable(sundials-addon-example)
```

Comment:
Should we comment this block out or tell users to comment it out unless they want to build the example addon?

Reply:
I think we can tell them to comment it out or guard it with another variable. It would be nice to have it uncommented so we can test the external addon functionality in our CI/nightly.